### PR TITLE
FEATURE: Add support for DLX

### LINKED ISSUE #5: Support for Dead Letter Exchanges
It should be possible to configure the RabbitQueue to use a DLX https://www.rabbitmq.com/dlx.html instead of the current requeueing mechanism.

Proposal: Add a configuration option to enable usage of a DLX.

If a DLX is in use, the `release` function could look like this:
```
public function release(string $messageId, array $options = []): void
{
    $this->channel->basic_nack($messageId);
}
```
This way, the message would not have to be requeued using the ReleaseService. Once that works, the `x-death` message header can be used to track the retry count of the message header.

Background:
I publish the jobs to a fanout exchange so multiple workers receive the same message. When a message fails, it is published to that exchange again, leading to duplicate messages in all worker queues.

### PR DESCRIPTION
**Pull request by paxuclus**
Closes #5

### CODE REVIEW

File: Classes/Queue/RabbitQueue.php [hunk 1]
```diff
@@ -258,7 +277,7 @@ protected function queue(string $payload, array $options = [], int $numberOfRele
 
         $headers = new AMQPTable($headerOptions);
         $message->set('application_headers', $headers);
-        $this->channel->basic_publish($message, $this->exchangeName, $this->routingKey !== '' ? $this->routingKey : $this->name);
+        $this->channel->basic_publish($message, $this->exchangeName, $this->routingKey);
```
Comment (paxuclus) [inline Classes/Queue/RabbitQueue.php:279]: This change might be breaking.

I think it should be possible to publish messages without a routing key. Maybe by adding a new configuration option for that?

Comment (johannessteu) [reply Classes/Queue/RabbitQueue.php:279]: Yeah, let's just release a major then?

**Review by johannessteu — CHANGES_REQUESTED** [review]
Could you please also document the dlx option in the Readme and `Settings.yaml` ? Otherwise i'd like to merge it :)

File: Classes/Queue/RabbitQueue.php (hunk 1 above)
Comment (paxuclus) [reply Classes/Queue/RabbitQueue.php:279]: Sure!

**Comment by paxuclus** [comment]
whoops, added the option to the README. 👍

**Review by johannessteu — APPROVED** [review]

**Comment by johannessteu** [comment]
Do you wanna test it with `dev-master` before i release a 2.0 ?

**Comment by paxuclus** [comment]
@johannessteu: I'm running my fork in semi-production already, so it's fine from my side. 😄

**Comment by johannessteu** [comment]
There you go :)
https://github.com/t3n/JobQueue.RabbitMQ/releases/tag/2.0.0

**Pull request merged**